tinyxml2.cpp

```diff
@@ -204,6 +205,7 @@ char* StrPair::ParseText( char* p, const char* endTag, int strFlags )
             return p + length;
         }
         ++p;
+        TIXMLASSERT( p );

```

Comment:
@leethomason I mean `p !=0` everywhere - both before the loop and on each iteration.